test/units/parsing/test_ajson.py

```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
     assert isinstance(data['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['bar']['baz'][0]['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['foo']['password'], AnsibleVaultEncryptedUnicode)
+
+
+def vault_data():
+    """
+    Prepares data for testing handling AnsibleVaultEncryptedUnicode
+    in AnsibleJSONEncoder.default()
+    """
+
+    with open(os.path.join(os.path.dirname(__file__), 'fixtures/ajson.json')) as f:
+        data = json.load(f, cls=AnsibleJSONDecoder)
+
+    data_0 = data['password']
+    data_1 = data['bar']['baz'][0]['password']
+
+    expected_0 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3339\n')
+
+    expected_1 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3338\n')
+
+    return dict(
+        data_0=data_0,
+        data_1=data_1,
+        expected_0=expected_0,
+        expected_1=expected_1,
+    )
+
+
+class TestAnsibleJSONEncoder():
+    """
+    Class for testing AnsibleJSONEncoder
```

Comment:
That's not true. This class is used only to namespace things. Not for testing.

Also, [stop writing classes](https://www.youtube.com/watch?v=o9pEzgHorH0). It's not worth it, not in this case.

Reply:
1. Good note about namespace.
2. Why should I stop writing classes? I do love them :)

Reply:
you can use modules for namespacing. using classes for this is overkill

Reply:
I change the class comment, thank you